runtime/queries/column_timeseries.go

```diff
@@ -0,0 +1,537 @@
+					continue
+				}
+				likeClause += escapedName + " " + prefix + " ILIKE '" + EscapeSingleQuotes(lv.GetStringValue()) + "'"
+				if j < len(dv.Like)-1 {
+					likeClause += " OR "
+				}
+			}
+			conditions = append(conditions, likeClause)
+		}
+		result += strings.Join(conditions, " "+dimensionJoiner+" ")
+		if i < len(dimensionValues)-1 {
+			result += ") AND ("
+		}
+	}
+	result += " ) "
+
+	return result
+}
+
+func getFilterFromMetricsViewFilters(filters *runtimev1.MetricsViewRequestFilter) string {
+	includeFilters := getFilterFromDimensionValuesFilter(filters.Include, "", "OR")
+	excludeFilters := getFilterFromDimensionValuesFilter(filters.Exclude, "NOT", "AND")
+	if includeFilters != "" && excludeFilters != "" {
+		return " ( " + includeFilters + ") AND (" + excludeFilters + ")"
+	} else if includeFilters != "" {
+		return includeFilters
+	} else if excludeFilters != "" {
+		return excludeFilters
+	} else {
+		return ""
+	}
+}
+
+func getFallbackMeasureName(index int, sqlName string) string {
+	if sqlName == "" {
+		s := fmt.Sprintf("measure_%d", index)
+		return s
+	} else {
+		return sqlName
+	}
+}
+
+var countName = "count"
+
+func normaliseMeasures(measures []*runtimev1.GenerateTimeSeriesRequest_BasicMeasure) []*runtimev1.GenerateTimeSeriesRequest_BasicMeasure {
+	if len(measures) == 0 {
+		return []*runtimev1.GenerateTimeSeriesRequest_BasicMeasure{
+			{
+				Expression: "count(*)",
+				SqlName:    countName,
+				Id:         "",
+			},
+		}
+	}
+	for i, measure := range measures {
+		measure.SqlName = getFallbackMeasureName(i, measure.SqlName)
+	}
+	return measures
+}
+
+const IsoFormat string = "2006-01-02T15:04:05.000Z"
+
+func sMap(k string, v float64) map[string]float64 {
+	m := make(map[string]float64, 1)
+	m[k] = v
+	return m
+}
+
+func convertToDateTruncSpecifier(specifier runtimev1.TimeGrain) string {
+	switch specifier {
+	case runtimev1.TimeGrain_TIME_GRAIN_MILLISECOND:
+		return "MILLISECOND"
+	case runtimev1.TimeGrain_TIME_GRAIN_SECOND:
+		return "SECOND"
+	case runtimev1.TimeGrain_TIME_GRAIN_MINUTE:
+		return "MINUTE"
+	case runtimev1.TimeGrain_TIME_GRAIN_HOUR:
+		return "HOUR"
+	case runtimev1.TimeGrain_TIME_GRAIN_DAY:
+		return "DAY"
+	case runtimev1.TimeGrain_TIME_GRAIN_WEEK:
+		return "WEEK"
+	case runtimev1.TimeGrain_TIME_GRAIN_MONTH:
+		return "MONTH"
+	case runtimev1.TimeGrain_TIME_GRAIN_YEAR:
+		return "YEAR"
+	}
+	panic(fmt.Errorf("unconvertable time grain specifier: %v", specifier))
+}
+
+func convertRowsToTimeSeriesValues(rows *drivers.Result, rowLength int) ([]*runtimev1.TimeSeriesValue, error) {
+	results := make([]*runtimev1.TimeSeriesValue, 0)
+	defer rows.Close()
+	var converr error
+	for rows.Next() {
+		value := runtimev1.TimeSeriesValue{}
+		results = append(results, &value)
+		row := make(map[string]interface{}, rowLength)
+		err := rows.MapScan(row)
+		if err != nil {
+			return results, err
+		}
+		value.Ts = row["ts"].(time.Time).Format(IsoFormat)
+		delete(row, "ts")
```

Comment:
Shouldn't this pass in and use `tsAlias`?

Reply:
No, because in the final output we replace `tsAlias` name to `ts`, please see 
```
        -- does not have that value.
        SELECT 
          ` + getCoalesceStatementsMeasures(measures) + `,
          template.` + tsAlias + ` as ts from template
        LEFT OUTER JOIN series ON template.` + tsAlias + ` = series.` + tsAlias + `
        ORDER BY template.` + tsAlias + `
```

Reply:
What if `getCoalesceStatementsMeasures` contains a measure named `ts`?